front/src/common/authorization/hooks/useUserResourceGrants.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from 'react';
+
+import {
+  mockedEditoastApi,
+  type GetGrantsByResourceTypeResponse,
+  type ResourceType,
+} from 'common/api/mock/mockEditoastApi';
+
+type ResourceGrants = Partial<Record<ResourceType, GetGrantsByResourceTypeResponse>>;
+
+/**
+ * provides the user grants on a list of resources
+ * @param payload the resource type and the resource ids
+ * @returns the user grants on the resources
+ */
+const useUserResourceGrants = (payload: Partial<Record<ResourceType, number[]>>) => {
+  const [fetchUserGrantByResourceId, { data: userResourcesGrants }] =
+    mockedEditoastApi.endpoints.postUserResourcesGrants.useMutation();
+
+  const resourceTypes = Object.keys(payload) as ResourceType[];
+
+  const resourceGrants = resourceTypes.reduce((acc, resourceType) => {
```

Comment:
To avoid the cast l.29
```suggestion
  const resourceGrants = resourceTypes.reduce<ResourceGrants>((acc, resourceType) => {
```